### ISSUE TITLE
support hashicorp Vault Secrets

### ISSUE THREAD

**Issue by fawaf** [external-secrets/external-secrets#3070]
**Is your feature request related to a problem? Please describe.**
support hashicorp Vault Secrets

**Describe the solution you'd like**
support hashicorp Vault Secrets

**Describe alternatives you've considered**
use some other secrets provider

**Additional context**
not sure if this feature request was already suggest, but i did not seem to find it looking through previously created issues.

**Comment by blakepettersson** [comment]
We use Vault in production and can assure you that it works as a charm

**Comment by Skarlso** [comment]
I think there are two hashicorp vault products, right? HashiCorp Vault and HashiCorp Vault Secrets. https://developer.hashicorp.com/hcp/docs/vault-secrets

**Comment by Skarlso** [comment]
I think what we support is HCP Vault and not HCP Vault Secrets. But please correct me if I'm wrong.

**Comment by fawaf** [comment]
yep, that is my understanding as well. hence the creation of this issue.

**Comment by evrardj-roche** [comment]
I am not very familiar with the differences between vault and HCP vault secrets.

If they are different: Patches are welcomed. Are you willing to work on this @fawaf ? 
See an example of new provider https://github.com/external-secrets/external-secrets/pull/5398 . 

If they are the same: We can probably close this :)

**Comment by fawaf** [comment]
> I am not very familiar with the differences between vault and HCP vault secrets.
> 
> If they are different: Patches are welcomed.

they are very different yeah.

> Are you willing to work on this [@fawaf](https://github.com/fawaf) ? See an example of new provider [#5398](https://github.com/external-secrets/external-secrets/pull/5398) .
> 
> If they are the same: We can probably close this :)

for sure, definitely. at some point :)

**Comment by alekc** [comment]
I don't think its worth implementing. From https://www.ibm.com/support/pages/hcp-vault-secrets-end-life

> HashiCorp will be decommissioning HCP Vault Secrets. The team is shifting focus to bringing the ease of use learnings from HCP Vault Secrets to HCP Vault Dedicated. Current HCP Vault Secrets users are encouraged to migrate to HCP Vault Dedicated.
> 
> Important Timelines
> End of Sale: June 30, 2025
> 
> New customers will no longer have access to HCP Vault Secrets
> Existing customers will be able to add new secrets to their HCP Vault Secrets applications until the end of life
> End of life: Earlier of the Flex contract expiry or July 1, 2026
> 
> HCP Vault Secrets applications will be deleted
> End of life and end of support

We can probably close this one off
